source/unsafe-messenger.js

```diff
@@ -73,6 +75,35 @@ export default function unsafeMessenger() {
 			editor.blur();
 		});
 	}
+
+	function monacoEditor(target) {
+		const editor = monaco.editor.getModels()[0];
+		let currentValue = editor.getValue();
+		sendBack(target, currentValue);
+
+		editor.onDidChangeContent((e) => {
+			// Check if the change was made by the user
+			if (e.isUserChange) {
+				// Check if the current value is different from the previous value
+				if (currentValue !== editor.getValue()) {
+					currentValue = editor.getValue();
+					throttle(50, sendBack)(target, currentValue);
+				}
+			}
+		});
+	
+		target.addEventListener('gt:transfer', () => {
+			editor.setValue(target.getAttribute('gt-value'));
+		});
+		target.addEventListener('gt:blur', () => {
+			if (editor.container.contentWindow) {
```

Comment:
What is this checking? When does it have contentWindow vs when does it have just a document?

Reply:
I just saw the earlier code. I'd make sure "Monaco in a frame" works at all or that it needs this, because GhostText already has its own frame management (testable on the demo page)

Reply:
Note: unless you saw where this is necessary (please provide a link if so) it should be dropped.

I think we only need:

```js
target.ownerDocument.activeElement.blur();
```